packages/flutter/lib/src/material/tabs.dart

```diff
@@ -381,19 +381,61 @@ class _TabsScrollBehavior extends BoundedBehavior {
   }
 }
 
-class TabBarSelection {
-  TabBarSelection({ int index: 0, this.maxIndex, this.onChanged }) : _index = index {
+class _TabBarSelection extends InheritedWidget {
+  _TabBarSelection({
+    this.selection,
+    Key key,
+    Widget child
+  }) : super(key: key, child: child);
+
+  final TabBarSelectionState selection;
+
+  bool updateShouldNotify(_TabBarSelection oldWidget) => selection != oldWidget.selection;
```

Comment:
I'm surprised this works. Surely the State is the same each time the TabBarSelectionState rebuilds.

I would have expected this to have the actual state, not the State (if that makes sense). That is, whatever values it is that decide whether we should notify that things need to rebuild.